src/sparcscore/pipeline/workflows.py

```diff
@@ -724,20 +724,24 @@ def cellpose_segmentation(self, input_image):
                     nucleus_cytosol_pairs[nucleus_id] = 0
             else:
                 continue
-        
-        # code to update the segmentation and nucleus masks with the lookup table here
-        # CODE HERE
-        # CODE HERE                       
-                 
-        #first when the masks are finalized save them to the maps
+
+        # now we have all the nucleus cytosol pairs we can filter the masks
+        for nucleus_id, cytosol_id in nucleus_cytosol_pairs.items():
+            if cytosol_id == 0:
+                masks_nucleus = np.where(masks_nucleus == nucleus_id, 0, masks_nucleus)
+                masks_cytosol = np.where(masks_cytosol == cytosol_id, 0, masks_cytosol)
+            else:
+                # set the cytosol_id in the cytosol mask to the id that the nucleus has:
+                masks_cytosol = np.where(masks_cytosol == cytosol_id, nucleus_id, masks_cytosol)
```

Comment:
Here I think we will potentially run into problems. Lets say we have the following lookup table:

[1:5,
3:1]

if we iterate through this we  will first change all px with value 5 to 1 in the cytosol mask. Great, this is what we want. But in the next step we then change all of the px with value 1 to 3. This is now unfortunately not only those which should belong to the cell with nucleusid 3 but also those that we just changed from 5 to 3. 
So I think we need to have a second condition where we check that we only update pxs that we have not already updated.

Reply:
Ohh yes, I missed it! Thank you!